src/index.test.ts

```diff
@@ -72,302 +49,125 @@ describe('BuxClient class', () => {
   });
 });
 
-describe('GetXpub', () => {
-  const HttpUrl = `/xpub`;
-  test('result', async () => {
-    // @ts-ignore
-    fetchMock.mockIf(/^.*$/, mockResponse(xpubJSON, HttpUrl, "xpub"));
 
-    await runTests(testClient, async (buxClient: TransportService) => {
-      const xPub = await buxClient.GetXPub();
-      expect(typeof xPub).toBe('object');
-      expect(xPub).toStrictEqual(JSON.parse(xpubJSON));
-    });
-  });
-});
-
-describe('UpdateXPubMetadata', () => {
-  const HttpUrl = `/xpub`;
-  const metadata: Metadata = {
-    "test-key": "test-value",
+describe('BuxClient routing', () => {
+  const options: ClientOptions = {
+    xPrivString: testClient.xPrivString,
+    signRequest: true,
   };
-  test('result', async () => {
-    // @ts-ignore
-    fetchMock.mockIf(/^.*$/, mockResponse(xpubJSON, HttpUrl, "xpub_metadata"));
+  const bux = new BuxClient(testClient.serverURL, options);
 
-    await runTests(testClient, async (buxClient: TransportService) => {
-      const xPub = await buxClient.UpdateXPubMetadata(metadata);
-      expect(typeof xPub).toBe('object');
-      expect(xPub).toStrictEqual(JSON.parse(xpubJSON));
-    });
-  });
-});
+  const tcs = [
```

Comment:
I would separate tests. When some tests are failed we need to make changes and run only tests which failed. Using loop it always runs all tests.

Reply:
I decided to apply the table-driven test approach because of the following advantages:

 *   I could logically group test cases.
 *   I avoided using obscure functions like "runTests," which can reduce code readability.
 *   I introduced clear and understandable code for the tests themselves.

Of course, I also avoided a significant amount of boilerplate code.

In case of failure of a specific test case, we receive precise information about which test case is causing the issue.

Additionally, these tests are quite fast, and the mentioned benefits carry much more weight and value than the ability to run tests for individual test cases only.

Reply:
I agree with advantages and have a big disadvantage. If for example `GetXpub` test fails need to run all tests in BuxClient routing group to check if it works after changes. In my case I need ~ 50 sec instead of ~1.5 sec.
<img width="379" alt="tests_time" src="https://github.com/BuxOrg/js-buxclient/assets/150656925/d01c5da1-d26a-4b61-b451-ac7dccaa4385">